### PR TITLE
grpc: add base add-on

### PR DESCRIPTION
**Pull request by amitpanwar789**
## Overview
This is the first PR for grpc support project.

## Related Issues
Related to https://github.com/zaproxy/zaproxy/issues/7695
@psiinon @thc202 
## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [ ] Squash commits
- [ ] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/grpc/grpc.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+description = "Inspect, attack Grpc endpoints and Decode protobuf message."
+
+zapAddOn {
+    addOnName.set("Grpc Support")
```
Comment (kingthorin) [inline addOns/grpc/grpc.gradle.kts:4]: gRPC

File: addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/contents/grpc.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
+<HTML>
+<HEAD>
+    <TITLE>
+        Grpc Support
```
Comment (kingthorin) [inline addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/contents/grpc.html:5]: gRPC

File: addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/contents/grpc.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    </TITLE>
+</HEAD>
+<BODY>
+<H1>Grpc Support</H1>
```
Comment (kingthorin) [inline addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/contents/grpc.html:9]: gRPC

File: addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/contents/grpc.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+This add-on will allow you to Inspect, attack Grpc endpoints and Decode protobuf message. It is currently under development.
```
Comment (kingthorin) [inline addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/contents/grpc.html:10]:
```suggestion
This add-on will allow you to inspect, attack gRPC endpoints, and decide the protobuf messages. It is currently under development.
```

File: addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/helpset.hs [hunk 5]
```diff
@@ -0,0 +1,41 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE helpset
+  PUBLIC "-//Sun Microsystems Inc.//DTD JavaHelp HelpSet Version 2.0//EN"
+         "http://java.sun.com/products/javahelp/helpset_2_0.dtd">
+<helpset version="2.0" xml:lang="en-GB">
+  <title>Grpc Support Add-On</title>
```
Comment (kingthorin) [inline addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/helpset.hs:6]: gRPC

File: addOns/grpc/src/main/resources/org/zaproxy/addon/grpc/resources/Message.properties [hunk 6]
```diff
@@ -0,0 +1,2 @@
+grpc.desc = Allows you to Inspect, attack Grpc endpoints and Decode protobuf message.
+grpc.name = Grpc Support.
```
Comment (kingthorin) [inline addOns/grpc/src/main/resources/org/zaproxy/addon/grpc/resources/Message.properties:2]: gRPC

File: addOns/grpc/src/main/resources/org/zaproxy/addon/grpc/resources/Message.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+grpc.desc = Allows you to Inspect, attack Grpc endpoints and Decode protobuf message.
```
Comment (kingthorin) [inline addOns/grpc/src/main/resources/org/zaproxy/addon/grpc/resources/Message.properties:1]:
```suggestion
grpc.desc = Allows you to inspect, attack gRPC endpoints, and decode protobuf messages.
```

File: addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/toc.xml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?xml version='1.0' encoding='UTF-8' ?>
+<!DOCTYPE toc
+        PUBLIC "-//Sun Microsystems Inc.//DTD JavaHelp TOC Version 2.0//EN"
+        "http://java.sun.com/products/javahelp/toc_2_0.dtd">
+
+<toc version="2.0">
+    <tocitem text="ZAP User Guide" tocid="toplevelitem">
+        <tocitem text="Add Ons" tocid="addons">
+            <tocitem text="Grpc Support" image="addon.grpc.icon" target="addon.grpc"/>
```
Comment (kingthorin) [inline addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/toc.xml:9]: gRPC

File: addOns/grpc/grpc.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+description = "Inspect, attack Grpc endpoints and Decode protobuf message."
```
Comment (kingthorin) [inline addOns/grpc/grpc.gradle.kts:1]:
```suggestion
description = "Inspect, attack gRPC endpoints, and decode protobuf messages."
```

File: addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/map.jhm [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE map
+        PUBLIC "-//Sun Microsystems Inc.//DTD JavaHelp Map Version 1.0//EN"
+        "http://java.sun.com/products/javahelp/map_1_0.dtd">
+
+<map version="1.0">
+    <mapID target="addon.grpc.icon" url="contents/images/grpc.svg" />
```
Comment (kingthorin) [inline addOns/grpc/src/main/javahelp/org/zaproxy/addon/grpc/resources/help/map.jhm:7]: png?

Drop the trailing space

**Review by kingthorin — APPROVED** [review]

File: addOns/grpc/grpc.gradle.kts [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    zapVersion.set("2.14.0")
+
+    manifest {
+        author.set("ZAP Dev Team")
+        url.set("https://www.zaproxy.org/docs/desktop/addons/grpc-support/")
+    }
+
+    apiClientGen {
+        messages.set(file("src/main/resources/org/zaproxy/addon/grpc/resources/Messages.properties"))
+    }
```
Comment (thc202) [inline addOns/grpc/grpc.gradle.kts:14]: Preferable not to add at this point, there's no API yet.

File: addOns/grpc/src/main/resources/org/zaproxy/addon/grpc/resources/Message.properties [hunk 12]
```diff
@@ -0,0 +1,2 @@
+grpc.desc = Allows you to inspect, attack gRPC endpoints, and decode protobuf messages.
+grpc.name = gRPC Support.
```
Comment (thc202) [inline addOns/grpc/src/main/resources/org/zaproxy/addon/grpc/resources/Message.properties:2]: Remove dot.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**